nomos-da/network/core/src/protocols/sampling/behaviour.rs

```diff
@@ -132,13 +143,64 @@ impl Clone for SamplingError {
                 peer_id: *peer_id,
                 error: *error,
             },
+            SamplingError::InvalidBlobId { blob_id, peer_id } => SamplingError::InvalidBlobId {
+                peer_id: *peer_id,
+                blob_id: blob_id.clone(),
```

Comment:
Nitpick: maybe is bacause of generics, but real blob_id is copy.

Reply:
The types differ for these invalid blob errors, they are Vec<u8> because this is what we get from the net. In the other errors, that come from the inside the node, the type is [u8;32]. If we'll move away from protobufs, maybe this will be solved.